### PR TITLE
Support azp and mutliple aud

### PR DESCRIPTION
**Pull request by damienbod**

### CODE REVIEW

**Comment by damienbod** [comment]
fixes https://github.com/damienbod/angular-auth-oidc-client/issues/582

File: projects/angular-auth-oidc-client/src/lib/validation/token-validation.service.ts [hunk 1]
```diff
@@ -261,6 +262,28 @@ export class TokenValidationService {
...
+            if (!dataIdToken.azp) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+    // If an azp (authorized party) Claim is present, the Client SHOULD verify that its client_id is the Claim Value.
+    validateIdTokenAzpValid(dataIdToken: any, clientId: string): boolean {
+        if (!dataIdToken.azp) {
```
Comment (FabianGosebrink) [inline projects/angular-auth-oidc-client/src/lib/validation/token-validation.service.ts:276]: add a questionmark --> ` if (!dataIdToken?.azp) {` checks for null and the `azp` property

File: projects/angular-auth-oidc-client/src/lib/validation/token-validation.service.ts [hunk 2]
```diff
@@ -261,6 +262,28 @@ export class TokenValidationService {
         return true;
     }
 
+    validateIdTokenAzpExistsIfMoreThanOneAud(dataIdToken: any, aud: any): boolean {
+        if (dataIdToken.aud instanceof Array && dataIdToken.aud.length > 1) {
```
Comment (FabianGosebrink) [inline projects/angular-auth-oidc-client/src/lib/validation/token-validation.service.ts:266]: Turn nested ifs into early returns :)

File: projects/angular-auth-oidc-client/src/lib/validation/token-validation.service.ts [hunk 3]
```diff
@@ -242,11 +242,12 @@ export class TokenValidationService {
     // not trusted by the Client.
     validateIdTokenAud(dataIdToken: any, aud: any): boolean {
         if (dataIdToken.aud instanceof Array) {
```
Comment (FabianGosebrink) [inline projects/angular-auth-oidc-client/src/lib/validation/token-validation.service.ts:244]: Array.IsArray()

File: projects/angular-auth-oidc-client/src/lib/validation/token-validation.service.ts (hunk 1 above)
Comment (damienbod) [reply projects/angular-auth-oidc-client/src/lib/validation/token-validation.service.ts]: fixed

File: projects/angular-auth-oidc-client/src/lib/validation/token-validation.service.ts (hunk 3 above)
Comment (damienbod) [reply projects/angular-auth-oidc-client/src/lib/validation/token-validation.service.ts]: fixed

**Pull request merged**